changefeedccl: add integrity validator

asg0451 opened the pull request:
Add a new integrity validator to the cdctest package. This validator checks for gaps in data in a contrived workload.

Epic: CRDB-38755

Release note: none

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/126562)
<!-- Reviewable:end -->

asg0451 committed:
changefeedccl: add integrity validator

Add a new integrity validator to the cdctest package. This validator checks for gaps in data in a contrived workload.

Epic: CRDB-38755

Release note: none

asg0451 commented on pkg/cmd/roachtest/tests/cdc.go:
```diff
@@ -1573,6 +1576,52 @@ func registerCDC(r registry.Registry) {
+		Cluster:          r.MakeClusterSpec(6, spec.CPU(16)),
+		Leases:           registry.MetamorphicLeases,
+		CompatibleClouds: registry.AllExceptAWS,
+		Suites:           registry.Suites(registry.Nightly),
```
TODO: change this to manual only for now since it fails with the sarama kafka sink

asg0451 commented on pkg/cmd/roachtest/tests/cdc.go:
```diff
@@ -1573,6 +1576,52 @@ func registerCDC(r registry.Registry) {
 	})
+
+	r.Add(registry.TestSpec{
+		Name:             "cdc/kafka-chaos-integrity",
```
TODO: make other roachtests with this workload&validator (webhooks, ..?)
